### ISSUE TITLE
fix ":compact" show function

### ISSUE THREAD

**Issue by fchorney** [invenia/JLSO.jl#5]
The Base.show function for JSLO seems to have a typo https://github.com/invenia/JLSO.jl/blob/master/src/JLSO.jl#L107

I assume `:compat` should be `:compact`, but if that gets fixed then we get an infinite loop when trying to print the compact version.

```julia
show: Error During Test at /Users/fchorney/Documents/repos/julia/JLSO.jl/test/JLSO.jl:64
  Test threw exception
  Expression: sprint(show, jlso; context=:compact => true) == expected
  StackOverflowError:
  Stacktrace:
   [1] print(::IOContext{Base.GenericIOBuffer{Array{UInt8,1}}}, ::JLSOFile) at ./strings/io.jl:29
   [2] show(::IOContext{Base.GenericIOBuffer{Array{UInt8,1}}}, ::JLSOFile) at /Users/fchorney/Documents/repos/julia/JLSO.jl/src/JLSO.jl:105
   [3] print(::IOContext{Base.GenericIOBuffer{Array{UInt8,1}}}, ::JLSOFile) at ./strings/io.jl:31
   ... (the last 2 lines are repeated 13768 more times)
   [27540] show(::IOContext{Base.GenericIOBuffer{Array{UInt8,1}}}, ::JLSOFile) at /Users/fchorney/Documents/repos/julia/JLSO.jl/src/JLSO.jl:105
   [27541] #sprint#325(::Pair{Symbol,Bool}, ::Int64, ::Function, ::Function, ::JLSOFile) at ./strings/io.jl:99
   [27542] (::getfield(Base, Symbol("#kw##sprint")))(::NamedTuple{(:context,),Tuple{Pair{Symbol,Bool}}}, ::typeof(sprint), ::Function, ::JLSOFile) at ./none:0
   [27543] macro expansion at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1166 [inlined]
   [27544] macro expansion at /Users/fchorney/Documents/repos/julia/JLSO.jl/test/JLSO.jl:48 [inlined]
   [27545] macro expansion at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1083 [inlined]
   [27546] macro expansion at /Users/fchorney/Documents/repos/julia/JLSO.jl/test/JLSO.jl:40 [inlined]
   [27547] macro expansion at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1083 [inlined]
```

**Comment by iamed2** [comment]
Yeah `print` calls `show` if there's not `print` method defined (and there isn't). To maintain current behaviour you could just remove the if block.

**Comment by fchorney** [comment]
yeah that's fair. If somebody wants to define a compact format show later on then they can add one back in, as it seems like this wasn't necessarily set up with that in mind.

**Comment by fchorney** [comment]
Should I just mush this into #4

**Comment by fchorney** [comment]
Closed in #4
